### DISCUSSION TITLE
Descriptions for epidermises - show on icon click?

### DISCUSSION THREAD

**Discussion by mefistotelis** [swfans/syndwarsfx#116] [Ideas]
The game includes descriptions of Epidermis mods, but they can only be viewed by using cheats to enable their research.

Options for improvement:

1. Make descriptions reachable by clicking on skin icon

When an agent has a skin, it could be made possible to click on the skin icon to show its description, even it it is not researched.
(descriptions of other weapons and mods are not available until researched)

2. Modify the descriptions to talk about blueprints

In the post-release demo, Bullfrog modified the descriptions so they do not refer to "the skin", but instead to "blueprints to make the skin". If we choose the skin description to be only reachable when researched, then we should adopt this change from the demo.

Personally, I think option 1 - clickable epidermis icon - would be better.

**Comment by geist22** [comment]
Clicking on the icon to show the description sounds nice. Epidermis can't be researched (without cheats) so I don't see a need to change description.

By the way, did anyone ever find out if there's a Epidermis 4 in the Church campaign? I have never found one myself. Many walkthroughs seem to have misinformation about it, like using the levels from the Playstation version. Or that the Epidermis 3 is in the warehouse with the weapons (there is no item pickup for 3 anywhere, you automatically get it awarded for one agent just for completing the level).

**Comment by mefistotelis** [comment]
> did anyone ever find out if there's a Epidermis 4 in the Church campaign?

So Church Cape Town is mission 45, now `-m 0,45`. No epidermis. But look at `-m 3,70`.

Unless there was a beta version released before this level got replaced, it looks like the info on location of Epidermises comes from developers rather than from players.

**Comment by geist22** [comment]
> But look at -m 3,70.

Interesting!

So this is one of those many unfinished maps that were later replaced with the final version used in the release, I take it? And they either choose not to place the Epidermis there, or just forgot about it.

**Comment by Moburma** [comment]
It's a really weird pair of maps. the "early" version C044L003.D1 actually has one of the last timestamps of any map in the game and was written to two months after the final mission 45 level c044l002.dat. It has exactly the same number of enemies in the same places (c044l002.dat has them slightly toughened up, health wise). The big clincher though is the commands list,  C044L003.D1 is far simpler than c044l002.dat. The final one has all kinds of checking if units are still alive, protection commands for other units, all sorts going on. The early one is just basic "kill group X". The final one has 351 commands vs a mere 134. On that alone I would say C044L003.D1 is an earlier unfinished version.
Maybe with deadlines looming they realised they hadn't put the church l4 epidermis in, loaded the wrong version of the map, added it, and that's why the file stamp is so much later?
My level editor is working pretty good now so I could easily add it to the final one.

EDIT: In fact, here is a link to  c044l002.dat with the skin added: https://filebin.net/0h13fso7tqubzqs7

Getting on topic, I personally vote for option 1, I think these should remain secret bonuses for high level players (that can't be made), but it would be cool to see the description. It is interesting that it looks like they toyed with making it buildable, in the pre-alpha it looks like they were going to be standard items, not special rewards.

**Comment by geist22** [comment]
> In fact, here is a link to c044l002.dat with the skin added

Thanks, I'll be using that. Although I have mixed feelings about "non-canon" content (especially giving yourself free stuff), I think Bullfrog just forgot to add it in this case.

I wonder why some walkthroughs refer to it when it's in a prototype map but not in the release version. Were the files datamined or did some developers actually write guides or gave it away in interviews? Would be interesting to know the source of this.

**Comment by mefistotelis** [comment]
> I wonder why some walkthroughs refer to it when it's in a prototype map

The reason provided by @Moburma seem the most plausible. At final stages of production there was a decision that this item will not be researchable but a bonus, so changes in code and in maps were made accordingly, but wrong version of this particular map was modified.

And after the game was released, some team members were active on certain bulletin boards, answering questions about the game. Game magazines were publishing tips as well - it's unlikely they mined game data these by themselves. I've also seen info on Mike Diskett answering emails, placed on some old fan sites.

**Comment by Moburma** [comment]
This bugged me - where did people get this idea of where the items were supposed to be? Then I realised the answer was almost literally in front of me when I saw my copy of the official strategy guide. It's from there, isn't it!

The third church skin is quite interesting as well, there's a nuance to where it is supposed to be I've not seen mentioned online:

![](https://i.ibb.co/nBxvmQX/cmiss18.png)

So you're supposed to BLOW UP the weapon storage room (which does exist on the map). This is total nonsense and there is nothing there in the actual level file, but maybe there are earlier versions of it where this is the case, I'll look later. 

Then of course we have the actual missing item:

![](https://i.ibb.co/WtJkDNs/cmiss21.png)

This matches up with the early file's placement. Annoyingly neither of these are specifically marked on the maps in the guide, you just get those descriptions. But for me it's confirmation Bullfrog really did intend the item to go there and it's a mistake that it's not in the game.
I will upload the version I made above to the SWLevels repository so people can restore this if they prefer.

Another odd thing I notice in the strategy guide is they also claim you have to blow up a building to get the first church epidermis. This is just plain wrong, and I think doing so will actually destroy the item. It also says there is Psycho gas hidden in the first EC level. 
No there isn't.

**Comment by geist22** [comment]
Good find. The writer of the official strategy guide credits some of the game's developers as sources of information.

I think "official" guides that receive their information from the publisher or even developers themselves are often made before the games even release, so if something in the game is changed in the late stages of development, those changes might not make it in the guides.

**Comment by Moburma** [comment]
@mefistotelis Do you know exactly where you got the demo with the new epidermis and chromotap text? I checked two versions of the post release demo and can't see these changes, looks like it's something a bit rarer.

**Comment by mefistotelis** [comment]
The change was in latest demo, the "net demo".

Here are the WADs, and extracted POTs (the POTs are not suitable for any use, I only made the extraction good enough for them to generate, and assumed any further comparisons or copy-pasting will be manual)

[demo_wads.zip](https://github.com/swfans/swars/files/13806107/demo_wads.zip)
